source/common/grpc/context_impl.cc

```diff
@@ -32,8 +34,22 @@ Stats::StatName ContextImpl::makeDynamicStatName(absl::string_view name) {
   return stat_name;
 }
 
+// Gets the stat prefix and underlying storage, depending on whether request_names is empty
+std::pair<Stats::StatName, Stats::SymbolTable::StoragePtr>
```

Comment:
you can just return the StoragePtr here...the StatName is trivially constructed on the StoragePtr.

Think of StoragePtr as std::string, and StatName as the absl::string_view referencing the same string.

Reply:
The problem is that in one case, the StoragePtr is allocated in this function, and in the other case, the StatName is retrieved from a member variable that is a StatName. So in one case, I am returning ownership, and in the other case I am not. 

Given that requirement, can I simplify this somehow?

Reply:
I see, that makes sense. And in all cases we don't need a symbol-table lock to do this operation, which is nice.

I think I see an opportunity for a refactor, where we add a new class StatNameJoinPool which has a method:
 ` StatName join(const std::vector<StatName>& parts);`
that would simplify this call-site and also others. Basically every usage of SymbolTable::join() is kind of ugly.

I added https://github.com/envoyproxy/envoy/issues/10516 to track this, but feel free to mark this resolved.